recipe/meta.yaml

```diff
@@ -1,221 +1,130 @@
 {% set name = "xgboost" %}
-{% set version = "1.7.4" %}
+{% set xgboost_version = environ['XGBOOST_VERSION'] %}
+{% set cuda_version = '.'.join(environ['RAPIDS_CUDA_VERSION'].split('.')[:2]) %}
+{% set cuda_major = cuda_version.split('.')[0] %}
+{% set py_version = environ['CONDA_PY'] %}
+{% set xgboost_git_repo = environ['XGBOOST_GIT_REPO'] %}
+{% set xgboost_git_ref = environ['XGBOOST_GIT_REF'] %}
+{% set build_number = environ['XGBOOST_BUILD_NUMBER'] %}
+{% set rapids_version = environ['RAPIDS_VERSION'] %}
 
 package:
-  name: {{ name|lower }}-split
-  version: {{ version }}
+  name: {{ name|lower }}
+  version: {{ xgboost_version }}
 
 source:
-  # we have to use git urls here to get the submodules needed for the
-  # build
-  git_url: https://github.com/dmlc/xgboost
-  git_tag: v{{ version }}
+  # We have to use git urls here to get the submodules needed for the build
+  git_url: {{ xgboost_git_repo }}
+  git_tag: {{ xgboost_git_ref }}
   patches:
-    # xgboost patches
-    - 0001-conda-Unbundle-libxgboost.-dll-dylib-so.patch
-    - 0001-Force-endian-flag-in-cross-compilation-mode.patch  # [arm64 or aarch64 or ppc64le]
+    - xgboost_master.patch
 
 build:
-  number: 2
-  skip: true  # [win and cuda_compiler_version != "None"]
-  skip: true  # [cuda_compiler_version == "10.2"]
+  number: {{ build_number }}
+  ignore_run_exports_from:
+    - {{ compiler('cuda') }}
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - {{ compiler('cuda') }}                 # [cuda_compiler_version != "None"]
-    - git                    # [not win]
-    - m2-git                 # [win]
+    - {{ compiler('cuda') }} ={{ cuda_version }}  # [xgboost_proc_type == "gpu"]
+    - git
     - cmake
-    - llvm-openmp  # [osx]
-    - ninja
+    - make
+    - librmm ={{ rapids_version }}
   host:
-    - cudatoolkit      # [(cuda_compiler_version or "").startswith("11")]
-    - nccl       # [cuda_compiler_version != "None"]
-    - llvm-openmp  # [osx]
+    - cudatoolkit {{ cuda_version }}
+    - nccl >=2.5  # [xgboost_proc_type == "gpu"]
+    - librmm ={{ rapids_version }}
+
 
 outputs:
+  - name: xgboost-proc
+    version: 1.0.0
+    build:
+      number: 0
+      string: "{{ xgboost_proc_type }}"
+      ignore_run_exports_from:
+        - {{ compiler('cuda') }}
+    about:
+      home: https://github.com/conda-forge/xgboost-feedstock
+      license: BSD-3-Clause
+      license_family: BSD
+      summary: A meta-package to select CPU or GPU build.
+
   - name: libxgboost
-    script: install-libxgboost.sh  # [not win]
-    script: install-win-wrapper.bat  # [win]
+    script: install-libxgboost.sh
     build:
-      activate_in_script: True
-      string: cuda{{ cuda_compiler_version | replace('.', '') }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version != "None"]
-      string: cpu_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version == "None"]
+      string: cuda{{ cuda_major }}_{{ build_number }}
```

Comment:
As this upstream `build/string`s are more complete, might be better to prefer those here